parity/deprecated.rs

```diff
@@ -21,93 +21,88 @@ use cli::Args;
+			Deprecated::Removed(s) => write!(f, "Option '{}' has been removed and is no longer supported.", s)
 		}
 	}
 }
 
-impl Deprecated {
-	fn jsonrpc() -> Self {
-		Deprecated::DoesNothing("--jsonrpc")
-	}
+pub fn find_deprecated(args: &Args) -> Vec<Deprecated> {
+	let mut result = vec![];
 
-	fn rpc() -> Self {
-		Deprecated::DoesNothing("--rpc")
+	if args.flag_jsonrpc {
+		result.push(Deprecated::DoesNothing("--jsonrpc"));
 	}
 
-	fn jsonrpc_off() -> Self {
-		Deprecated::Replaced("--jsonrpc-off", "--no-jsonrpc")
+	if args.flag_rpc {
+		result.push(Deprecated::DoesNothing("--rpc"));
 	}
 
-	fn webapp() -> Self {
-		Deprecated::DoesNothing("--webapp")
+	if args.flag_jsonrpc_off {
+		result.push(Deprecated::Replaced("--jsonrpc-off", "--no-jsonrpc"));
 	}
 
-	fn dapps_off() -> Self {
-		Deprecated::Replaced("--dapps-off", "--no-dapps")
+	if args.flag_webapp {
+		result.push(Deprecated::DoesNothing("--webapp"));
 	}
 
-	fn ipcdisable() -> Self {
-		Deprecated::Replaced("--ipcdisable", "--no-ipc")
+	if args.flag_dapps_off {
+		result.push(Deprecated::Replaced("--dapps-off", "--no-dapps"));
 	}
 
-	fn ipc_off() -> Self {
-		Deprecated::Replaced("--ipc-off", "--no-ipc")
+	if args.flag_ipcdisable {
+		result.push(Deprecated::Replaced("--ipcdisable", "--no-ipc"));
 	}
 
-	fn etherbase() -> Self {
-		Deprecated::Replaced("--etherbase", "--author")
+	if args.flag_ipc_off {
+		result.push(Deprecated::Replaced("--ipc-off", "--no-ipc"));
 	}
 
-	fn extradata() -> Self {
-		Deprecated::Replaced("--extradata", "--extra-data")
+	if args.flag_etherbase.is_some() {
+		result.push(Deprecated::Replaced("--etherbase", "--author"));
 	}
-}
-
-pub fn find_deprecated(args: &Args) -> Vec<Deprecated> {
-	let mut result = vec![];
 
-	if args.flag_jsonrpc {
-		result.push(Deprecated::jsonrpc());
+	if args.flag_extradata.is_some() {
+		result.push(Deprecated::Replaced("--extradata", "--extra-data"));
 	}
 
-	if args.flag_rpc {
-		result.push(Deprecated::rpc());
+	// Removed in 1.7
+	if args.flag_dapps_port.is_some() {
+		result.push(Deprecated::Replaced("--dapps-port", "--jsonrpc-port"));
 	}
 
-	if args.flag_jsonrpc_off {
-		result.push(Deprecated::jsonrpc_off());
+	if args.flag_dapps_interface.is_some() {
+		result.push(Deprecated::Replaced("--dapps-interface", "--jsonrpc-interface"));
 	}
 
-	if args.flag_webapp {
-		result.push(Deprecated::webapp())
+	if args.flag_dapps_hosts.is_some() {
+		result.push(Deprecated::Replaced("--dapps-hosts", "--jsonrpc-hosts"));
 	}
 
-	if args.flag_dapps_off {
-		result.push(Deprecated::dapps_off());
+	if args.flag_dapps_cors.is_some() {
+		result.push(Deprecated::Replaced("--dapps-cors", "--jsonrpc-cors"));
 	}
 
-	if args.flag_ipcdisable {
-		result.push(Deprecated::ipcdisable());
+	if args.flag_dapps_user.is_some() {
+		result.push(Deprecated::Removed("--dapps-user"));
 	}
 
-	if args.flag_ipc_off {
-		result.push(Deprecated::ipc_off());
+	if args.flag_dapps_pass.is_some() {
+		result.push(Deprecated::Removed("--dapps-pass"));
```

Comment:
How does json-rpc authentication work now? (I was not following your recent prs closely) :p